src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -543,30 +540,24 @@ class AssemblerConnectionManager(
         }
     }
 
-    fun getAllRoles(): BasePostgresIterable<Role> {
-        return BasePostgresIterable(PreparedStatementHolderSupplier(hds, PRINCIPALS_SQL) { ps ->
-            ps.setString(1, PrincipalType.ROLE.name)
-        }) {
-            securePrincipalsManager.getSecurablePrincipal(ResultSetAdapters.aclKey(it)) as Role
-        }
+    fun getAllRoles(): Set<Role> {
+        return securePrincipalsManager.allRoles
     }
 
-    fun getAllUsers(): BasePostgresIterable<SecurablePrincipal> {
```

Comment:
This now requires the list of all roles to fit in memory forever.

Reply:
that was already a requirement

Reply:
Without check who wrote the line, it looks like it was intended to page through role by role, instead of pulling all roles to a single box. This will work fine for a while until we have a lot of roles so it's not that big of an issue.

Reply:
yeah, but then we did a `.map` on them collecting them to a list 😝 

Anyway this should make things faster since this updates the logic to not do a separate hazelcast read for every role in the system anymore, but at some point we'll probably want to set up paging for reading out principals.